src/main/kotlin/lotto/Statistics.kt

```diff
@@ -1,35 +1,27 @@
 package lotto
 
 // here ticket is a constructor parameter, but not a property
-class Calculator(
+class Statistics(
     tickets: List<Lotto>,
+    purchaseAmount: Int,
     private val winningNumbers: WinningNumbers,
-    private var _results: MutableMap<Rank, Int> = emptyMap<Rank, Int>().toMutableMap(),
 ) {
-    init {
-        calculateResults(tickets)
-    }
-    val results: Map<Rank, Int>
-        get() = _results
+    val results: Map<Rank, Int> = calculateResults(tickets)
```

Comment:
Instead of creating internal variables, return the result directly through a method. 😊